certstore/certstore.go

```diff
@@ -389,8 +389,9 @@ func (cs *Store) Put(ctx context.Context, cert *certs.FinalityCertificate) error
 		return fmt.Errorf("putting the cert: %w", err)
 	}
 
-	if cert.GPBFTInstance%cs.powerTableFrequency == 0 {
-		if err := cs.putPowerTable(ctx, cert.GPBFTInstance, cs.latestPowerTable); err != nil {
+	// The new power table is the power table to validate the _next_ instance.
+	if (cert.GPBFTInstance+1)%cs.powerTableFrequency == 0 {
+		if err := cs.putPowerTable(ctx, cert.GPBFTInstance+1, newPowerTable); err != nil {
```

Comment:
The fact that this has been here for so long makes me slightly unsure of f3 with changing power table.

Reply:
So, the power table _diff_ was correct. The issue was that we were storing it at the wrong time. But yeah, the current tests _should_ catch this (I plan on looking into it today).